kyo-core/shared/src/main/scala/kyo/scheduler/IOPromise.scala

```diff
@@ -12,7 +13,7 @@ import scala.util.control.NoStackTrace
 
 private[kyo] class IOPromise[+E, +A](init: State[E, A]) extends Safepoint.Interceptor:
 
-    @volatile private var state: State[E, A] = init
+    private val state = new AtomicReference[State[Any, Any]](init)
```

Comment:
I couldn't find a supported way to perform atomic operations on a field in Native without allocating an `AtomicReference`. This will introduce a performance regression with a new allocation for each fiber. I'll create a ticket to follow up.